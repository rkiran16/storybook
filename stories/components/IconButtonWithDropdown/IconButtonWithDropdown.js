import { html } from "lit-html";

export const IconButtonWithDropdown = ({ textDirection, inverse = false, disabled = false, iconName, buttonText, className = 'p-3' }) => {
  const mode = inverse ? 'inverse' : '';
  const direction = textDirection === 'vertical' ? '' : 'horizontal flex-row-reverse';
  const modeBackground = inverse ? 'background-color:#27251F' : '';
  const textClass = buttonText ? 'btn-icon-w-text' : '';
  const spacingClass = textDirection === 'horizontal' ? 'ms-1' : '';
  const btnClass = `btn btn-icon ${textClass} ${direction} ${mode}`;

  return html`
    <div class=${`d-inline-block ${className}`} style=${modeBackground}>
      <div class="dropdown">
        <button
          type="button"
          ?disabled=${disabled}
          class=${`dropdown-toggle ${btnClass}`}
          id="dropdownIconButton" 
          data-bs-toggle="dropdown" 
          aria-expanded="false"
        >
          <span class="material-icons ${spacingClass}">${iconName}</span>
          ${buttonText}
        </button>
        <ul class="dropdown-menu dropdown-menu-end shadow-sm" aria-labelledby="dropdownIconButton">
          <li tabIndex="-1" class="dropdown-item">Action </li>
          <li tabIndex="-1" class="dropdown-item"><a tabIndex="-1" href="#">Link</a></li>
          <li tabIndex="-1" class="dropdown-item">Another Action</li>
        </ul>
      </div>
    </div>
  `;
};

export const IconButtonWithOrganizedDropdown = ({ textDirection, inverse = false, disabled = false, iconName, buttonText, className = 'p-3' }) => {
  const mode = inverse ? 'inverse' : '';
  const direction = textDirection === 'vertical' ? '' : 'horizontal flex-row-reverse';
  const modeBackground = inverse ? 'background-color:#27251F' : '';
  const textClass = buttonText ? 'btn-icon-w-text' : '';
  const spacingClass = textDirection === 'horizontal' ? 'ms-1' : '';
  const btnClass = `btn btn-icon ${textClass} ${direction} ${mode}`;

  return html`
    <div class=${`d-inline-block ${className}`} style=${modeBackground}>
      <div class="organized-dropdown dropdown">
        <button
          type="button"
          ?disabled=${disabled}
          class=${`dropdown-toggle ${btnClass}`}
          id="dropdownIconButton" 
          data-bs-toggle="dropdown" 
          aria-expanded="false"
        >
          <span class="material-icons ${spacingClass}">${iconName}</span>
          ${buttonText}
        </button>
        <ul class="dropdown-menu dropdown-menu-end shadow-sm" aria-labelledby="dropdownMenuButton">
          <li class="dropdown-header" tabindex="-1"><h6>Organized list header 1</h6></li>
          <li tabIndex="-1" class="dropdown-item">Action</li>
          <li tabIndex="-1" class="dropdown-item">Another Action</li>
          <li tabIndex="-1" class="dropdown-item"><a tabIndex="-1" href="#">Links work too</a></li>
          <li class="dropdown-header" tabindex="-1"><h6>Organized list header 2</h6></li>
          <li tabIndex="-1" class="dropdown-item">Action 2</li>
          <li tabIndex="-1" class="dropdown-item">Another Action 2</li>
          <li tabIndex="-1" class="dropdown-item"><a tabIndex="-1" href="#">Link 2</a></li>
        </ul>
      </div>
    </div>
  `;
};
