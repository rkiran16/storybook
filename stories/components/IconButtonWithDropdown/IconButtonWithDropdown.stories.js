import { IconButtonWithDropdown, IconButtonWithOrganizedDropdown } from './IconButtonWithDropdown';

export default {
  title: 'Components/Buttons/IconButtonWithDropdown',
  parameters: {
    layout: 'centered',
    zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/62f6596c524c4769bad620ac",
  },
  argTypes: {
    iconName: {
      name: "Icon",
      control: { type: 'select'},
      description: "Select Icon Name",
      defaultValue: "overflow",
      options: ['overflow', 'print', 'export'],
      mapping: {
        overflow: 'more_vert',
        print: 'print',
        export: 'logout'
      }
   },
   buttonText: {
      name: 'Button text',
      type: { name: 'string', required: false },
      defaultValue: 'Actions',
      description: 'Button Text',
      control: {
        type: 'text'
      }
   },
   textDirection: {
       name: "Text direction",
       control: { type: 'select'},
       description: "Alignment of text in relation to the icon",
       defaultValue: "vertical",
       options: ['vertical', 'horizontal'],
       table: {
        type: { summary: ['vertical', 'horizontal'] },
        defaultValue: { summary: 'vertical' },
      } 
    },
    inverse: {
      name: "Invert Colors",
      type: { name: 'boolean', required: false},
      description: "Invert colors",
      defaultValue: false,
      table: {
        type: { summary: 'boolean' },
        defaultValue: { summary: false },
      },
      control: {
        type: 'boolean'
      }
    },
    disabled: {
      name: "Disabled",
      type: { name: 'boolean', required: false},
      description: "Use when the user cannot proceed or until an input is collected",
      defaultValue: false,
      table: {
        type: { summary: 'boolean' },
        defaultValue: { summary: false },
      },
      control: {
        type: 'boolean'
      }
    }
  },
};

const Template = (args) => IconButtonWithDropdown(args);

export const Standard = Template.bind({});
Standard.parameters = {
  layout: 'centered',
};
Standard.args = {
  iconName: 'more_vert',
  buttonText: 'Actions',
  textDirection: 'vertical',
  disabled: false,
  inverse: false,
};

const OrganizedTemplate = (args) => IconButtonWithOrganizedDropdown(args);

export const Organized = OrganizedTemplate.bind({});
Organized.parameters = {
  layout: 'centered',
};
Organized.args = {
  iconName: 'more_vert',
  buttonText: 'Actions',
  textDirection: 'vertical',
  disabled: false,
  inverse: false,
};