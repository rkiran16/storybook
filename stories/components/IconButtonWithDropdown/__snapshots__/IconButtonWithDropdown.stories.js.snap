// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Buttons/IconButtonWithDropdown Organized 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div class=",
    " style=",
    ">
      <div class=\\"organized-dropdown dropdown\\">
        <button
          type=\\"button\\"
          ?disabled=",
    "
          class=",
    "
          id=\\"dropdownIconButton\\" 
          data-bs-toggle=\\"dropdown\\" 
          aria-expanded=\\"false\\"
        >
          <span class=\\"material-icons ",
    "\\">",
    "</span>
          ",
    "
        </button>
        <ul class=\\"dropdown-menu dropdown-menu-end shadow-sm\\" aria-labelledby=\\"dropdownMenuButton\\">
          <li class=\\"dropdown-header\\" tabindex=\\"-1\\"><h6>Organized list header 1</h6></li>
          <li tabIndex=\\"-1\\" class=\\"dropdown-item\\">Action</li>
          <li tabIndex=\\"-1\\" class=\\"dropdown-item\\">Another Action</li>
          <li tabIndex=\\"-1\\" class=\\"dropdown-item\\"><a tabIndex=\\"-1\\" href=\\"#\\">Links work too</a></li>
          <li class=\\"dropdown-header\\" tabindex=\\"-1\\"><h6>Organized list header 2</h6></li>
          <li tabIndex=\\"-1\\" class=\\"dropdown-item\\">Action 2</li>
          <li tabIndex=\\"-1\\" class=\\"dropdown-item\\">Another Action 2</li>
          <li tabIndex=\\"-1\\" class=\\"dropdown-item\\"><a tabIndex=\\"-1\\" href=\\"#\\">Link 2</a></li>
        </ul>
      </div>
    </div>
  ",
  ],
  "values": Array [
    "d-inline-block p-3",
    "",
    false,
    "dropdown-toggle btn btn-icon btn-icon-w-text  ",
    "",
    "more_vert",
    "Actions",
  ],
}
`;

exports[`Storyshots Components/Buttons/IconButtonWithDropdown Standard 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div class=",
    " style=",
    ">
      <div class=\\"dropdown\\">
        <button
          type=\\"button\\"
          ?disabled=",
    "
          class=",
    "
          id=\\"dropdownIconButton\\" 
          data-bs-toggle=\\"dropdown\\" 
          aria-expanded=\\"false\\"
        >
          <span class=\\"material-icons ",
    "\\">",
    "</span>
          ",
    "
        </button>
        <ul class=\\"dropdown-menu dropdown-menu-end shadow-sm\\" aria-labelledby=\\"dropdownIconButton\\">
          <li tabIndex=\\"-1\\" class=\\"dropdown-item\\">Action </li>
          <li tabIndex=\\"-1\\" class=\\"dropdown-item\\"><a tabIndex=\\"-1\\" href=\\"#\\">Link</a></li>
          <li tabIndex=\\"-1\\" class=\\"dropdown-item\\">Another Action</li>
        </ul>
      </div>
    </div>
  ",
  ],
  "values": Array [
    "d-inline-block p-3",
    "",
    false,
    "dropdown-toggle btn btn-icon btn-icon-w-text  ",
    "",
    "more_vert",
    "Actions",
  ],
}
`;
